HELM-341: Documentation formatting cleanup (#484)
OpenNMS/opennms-helm: docs/modules/getting_started/pages/basic_walkthrough.adoc

:!figure-caption:

[[getting-started-basic-walkthrough]]
= Basic Walk-Through

> This section demonstrates how to create a simple dashboard with {page-component-title}:

Reviewer: Note that line 24 should be "Set up plugin"

The walk-through assumes that you have:

> * an instance of Grafana with the {page-component-title} application installed

Author: ```suggestion
* An instance of Grafana with the {page-component-title} application installed.
```

* an instance of OpenNMS that is reachable from the Grafana instance above
* one or more nodes being monitored by OpenNMS

If you already use the OpenNMS Data Source plugin with Grafana, see xref:migrating_from_opennms_datasource.adoc#[Migrate from the OpenNMS data source plugin] for information on how to switch to {page-component-title}.

[[bw-app-setup]]
== Set up plugin

In this section, you will enable the {page-component-title} plugin in Grafana and create <<custom actions>> to help with alarm diagnosis.

. Navigate to the home page of your Grafana instance.

> . In the left menu, click the Configuration icon, and click *Plugins*:

Reviewer 2: ```suggestion
. In the left menu, click the *Configuration* icon, and click *Plugins*:
```

> * an instance of OpenNMS that is reachable from the Grafana instance above
> * one or more nodes being monitored by OpenNMS

Author: ```suggestion
* An instance of OpenNMS that is reachable from the Grafana instance above.
* One or more nodes being monitored by OpenNMS.
```

. In the Config tab, click *Enable*.
+

A *Disable* button indicates the plugin is already enabled.

> Skip to the next step (creating <<Custom actions>>).

Reviewer: ```suggestion
Skip to the next step (create <<custom actions>>).
```

To create a custom action, follow these steps:

. Navigate to the home page of your Grafana instance.

> . In the left menu, click on the Configuration icon and click *Plugins*.

Reviewer 2: ```suggestion
. In the left menu, click on the *Configuration* icon and click *Plugins*.
```

. Select *OpenNMS Helm*.
. On the Config tab, click *Add Custom Action*.
. Specify a label and enter a URL.
.. In most cases the URL will point to your OpenNMS instance (for example, `http://localhost:8980/opennms/`, appended with the `$variable` that references an OpenNMS alarm property).
. Click *Update*.

[[bw-ds-setup]]
== Datasource setup

The type of data you can display in your dashboard depends on the datasource you select.
Before you create a dashboard, you need to set up the datasource.
OpenNMS datasources include performance, flow, and entity.
For more information on OpenNMS datasources, see  xref:datasources:performance_datasource.adoc[Datasources].

In this section, you will configure the entity datasource.
Follow the same process to configure a flow or performance datasource.
The only fields you need to configure are the URL, Basic Auth, and Basic Auth Details (user and password) fields.
Refer to https://grafana.com/docs/grafana/latest/datasources/[Grafana's documentation] for information on configuring other fields.

. Navigate to the home page of your Grafana instance.

> . In the left menu, click on the Configuration icon, and click *Data Sources*:

Reviewer 2: ```suggestion
. In the left menu, click on the *Configuration* icon, and click *Data Sources*:
```

. Navigate to the home page of your Grafana instance.
. In the left menu, click on the Configuration icon and click *Plugins*:
. Select *OpenNMS Helm*.
. On the Config tab, click *Add Custom Action*:
. Specify a label and enter a URL.
.. In most cases the URL will be your OpenNMS instance, for example, `http://localhost:8980/opennms/`, appended with the $variable that references a property of an OpenNMS alarm.
. Repeat for each custom action you want to create.
. Click *Update*.

[[bw-ds-setup]]
== Datasource setup

The type of data you can display in your dashboard depends on the datasource you select.
Before you create a dashboard, you need to set up the datasource.
OpenNMS datasources include performance, flow, and entity.
For more information on OpenNMS datasources, see  xref:datasources:performance_datasource.adoc[Datasources].

> In this section, you will configure the Entity datasource.

Reviewer: ```suggestion
In this section, you will configure the entity datasource.
```

> Follow the same process to configure a Flow or Performance datasource.

Reviewer: ```suggestion
Follow the same process to configure a flow or performance datasource.
```

> . In the *Auth* area, click the toggle next to *Basic Auth*.

Reviewer 2: ```suggestion
. In the Auth area, click the toggle next to *Basic Auth*.
```

> 
> . Enter the credentials for an OpenNMS user in the *User* and *Password* fields.

Reviewer 2: ```suggestion
. Enter the credentials for an OpenNMS user in the User and Password boxes.
```

. Click *Save & Test*.
+
A success message should appear.
If the message indicates a failure, review the information you entered and click *Save & Test* to retest the datasource.

[[bw-dash-setup]]
== Dashboard setup

This section describes how to create a dashboard with an alarms table.
For information on displaying additional or different types of data, see xref:bw-panel-create[creating a panel].

. Create a new dashboard by clicking the *+* sign in the left sidebar to open the Create menu and choose *Dashboard*.
. Click *Add New Panel* and configure the panel in the lower half of the screen:
+
image::gf-panel-config.png[Configure Panel, 500]

. In the *Query* drop-down, select the datasource we previously created in xref:bw-ds-setup[Datasource Setup].

> 
> . Select *Alarms* in the *SELECT* drop-down.

Reviewer 2: ```suggestion
. Select *Alarms* in the Select list.
```

> .. Configure additional information and add filters as desired (click *+Query* to add queries).
> 

Reviewer 2: ```suggestion
.. Configure additional information and add filters as desired (click *+Query* to add queries).
```

> . In the *Panel* tab, add a name for the panel under *Settings > Panel*.

Reviewer 2: ```suggestion
. In the *Panel* tab, add a name for the panel under menu:Settings[Panel].
```

. Under *Visualization*, select *Alarm Table*.
The graph appears in the top of the panel.
.. You can further customize the graph by specifying information in the *Panel* tab.

. In the top-right menu, click *Save*.
. Enter a name for the dashboard and click *Save*.

That's it!
Now, you have your first dashboard.

Try changing the time range at the top-right of the screen, or right-click on an alarm to perform actions against it.

Create more panels to display other data, or xref:importing.adoc#gs-dashboard-import[import] an existing dashboard to learn more about how to design your own.

[[bw-panel-create]]
== Create a panel

A panel is the part of your dashboard that displays the fault and performance management data you specify.
Before you add a panel, determine the type of data you want to display (performance, metrics, alarms/nodes), and make sure that you have xref:bw-ds-setup[set up the associated datasource].
Default graph types in Grafana work only with the appropriate datasource (for example, an alarm table visualization will not display data if a flow datasource is specified).

> . In your dashboard, click the *Add Panel* icon and *Add New Panel*.
> 

Reviewer 2: ```suggestion
. In your dashboard, click the *Add Panel* icon and select *Add New Panel*.
```

> . Select the datasource from the *Query* list.
> 

Reviewer 2: ```suggestion
. Select the datasource from the *Query* list.
```

. Configure additional information as appropriate.

> Options are dependent on the type of datasource selected.
> 

Reviewer 2: ```suggestion
Options are dependent on the type of datasource selected.
```

> . In the *Visualization* area of the *Panel* tab, click the *Visualization* icon.

Reviewer 2: ```suggestion
. In the Visualization area of the *Panel* tab, click the *Visualization* icon.
```

. Click *Graph* or select another type of visualization (Stat, Gauge, Bar gauge).
+
The graph appears in the top of the panel.

.. Specify information in the bottom half of the panel to customize the graph.
. In the *Panel* tab, change the name of the panel and specify additional information.
. In the top-right, click *Save*.

[[bw-alarm-panel]]
== Create an alarm panel

An alarm panel displays alarms from an xref:datasources:entity_datasource.adoc#[entities datasource].
Visualization of entity datasource information (nodes and alarms) is normally done in tables.

This procedure creates an alarm panel using an alarm table visualization:

> . In your dashboard, click the *Add Panel* icon and select *Add New Panel*.
> 

Reviewer 2: ```suggestion
. In your dashboard, click the *Add Panel* icon and select *Add New Panel*.
```

> . Select an entity datasource from the *Query* list.
> 

Reviewer 2: ```suggestion
. Select an entity datasource from the *Query* list.
```

> . Choose Alarms from the *Select* list.

Reviewer 2: ```suggestion
. Choose *Alarms* from the *Select* list.
```

> 
> . Create filters and conditions to specify the nodes and alarms to display:

Reviewer 2: ```suggestion
. Create filters and conditions to specify the nodes and alarms to display:
```

.. Note that you can use nested "and/or" logic.
.. You can order alarms by ascending (oldest alarms first) or descending (most recent alarms).
.. You can limit the number of alarms displayed.
.. Toggle featured attributes on to display select attributes.
Toggle it off to display all attributes in the drop-down lists.
+
.Sample alarm table configuration
image::gf-alarm-table.png[Alarm table, 500]

> . In the *Visualization* area of the *Panel* tab, click the *Visualization* icon.

Reviewer 2: ```suggestion
. In the Visualization area of the *Panel* tab, click the *Visualization* icon.
```

. Select an *Alarm Table* graph.
The graph appears in the top of the panel.
.. You can choose table or histogram graphs for alarms.
For nodes, you can configure only a table graph.
.. You can further customize the graph (including alarm severity theme colors) in the bottom half of the screen and in other sections of the *Panel* tab.
. Change the name of the panel.

> . In the top-right menu, click *Save*.

Reviewer 2: ```suggestion
. In the top-right menu, click *Save*.

```

. Click *Save & Test*.
+
A success message should appear.
If the message indicates a failure, review the information you entered and click *Save & Test* to retest the datasource.

[[bw-dash-setup]]
== Dashboard Setup

This section describes how to create a dashboard with an alarm table.
For information on displaying additional or different types of data, see xref:bw-panel-create[creating a panel].

> . Create a new dashboard by clicking the *+* sign in the left menu and choosing *Create>Dashboard*.

Author: ```suggestion
. Create a new dashboard by clicking the *+* sign in the left sidebar to open the Create menu and choose *Dashboard*.
```

Create more panels to display other data, or xref:importing.adoc#gs-dashboard-import[import] an existing dashboard to learn more about how to design your own.

[[bw-panel-create]]
== Create a Panel
A panel is the part of your dashboard that displays the fault and performance management data you specify.
Before you add a panel, determine the type of data you want to display (performance, metrics, alarms/nodes), and make sure that you have xref:bw-ds-setup[set up the associated datasource].
Default graph types in Grafana work only with the appropriate data source (for example, an alarm table visualization will not display data if a flow datasource is specified.)

. In your dashboard, click the `Add panel` icon and *Add new panel*.

. In the *Query* drop-down, select the datasource.

. Configure additional information as appropriate.
Options are dependent on the type of datasource selected.

. In the *Visualization* area of the panel tab, click the visualization icon.
. Click *Graph* or select another type of visualization (Stat, Gauge, Bar gauge).
+
The graph appears in the top of the panel.

.. Specify information in the bottom half of the panel to customize the graph.

> . In *Panel* tab, change the name of the panel and specify additional information.

Reviewer: ```suggestion
. In the *Panel* tab, change the name of the panel and specify additional information.
```

> . In the top-right, click the *Save*.

Reviewer: ```suggestion
. In the top-right, click *Save*.
```

== Create an Alarm Panel
An alarm panel displays alarms from an xref:datasources:entity_datasource.adoc#[entities datasource].
Visualization of entity datasource information (currently nodes and alarms) is normally done in tables.

This procedure creates an alarm panel using an alarm table visualization.

. In your dashboard, click the `Add panel` icon  icon and *Add new panel*.

. In the *Query* drop-down, select an entity datasource.

. From *SELECT*, choose `Alarms`.

. Create filters and conditions to specify the nodes/alarms to display:
.. Note that you can use nested "and/or" logic.
.. You can order by ascending (oldest alarms first) or descending (most recent alarms) and limit the number of alarms displayed.
.. Toggle featured attributes on to display select attributes.
Toggle it off to display all attributes in the drop-down lists.

+
.Sample alarm table configuration
image::gf-alarm-table.png[Alarm table, 500]

. In the *Visualization* area of the panel tab, click the visualization icon.
. Select and *Alarm Table* graph.
.. With alarms, you can also choose `Table` or `Alarm histogram`; with nodes, choose `Table`
.. The graph appears in the top of the panel.

> .. You can further customize the graph by (including alarm severity theme colors) in the bottom half of the screen and in other sections of the Panel tab.

Reviewer: ```suggestion
.. You can further customize the graph (by including alarm severity theme colors) in the bottom half of the screen and in other sections of the *Panel* tab.
```
